diff -urN yasmin-0.6.0.org/Makefile yasmin-0.6.0/Makefile
--- yasmin-0.6.0.org/Makefile	Fri Feb 16 13:13:49 2001
+++ yasmin-0.6.0/Makefile	Fri Feb 16 13:42:44 2001
@@ -72,8 +72,8 @@
 SYSCONF_LINK_LIB_STATIC	= rm -f $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) ; \
 				 $(SYSCONF_AR) $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) $(OBJECTS) $(OBJMOC) 
 # Compiling application source
-SYSCONF_CXXFLAGS	= -pipe -DNO_DEBUG -O2  
-SYSCONF_CFLAGS		= -pipe -DNO_DEBUG -O2  
+SYSCONF_CXXFLAGS	= -pipe -DNO_DEBUG $(OPT_FLAGS)
+SYSCONF_CFLAGS		= -pipe -DNO_DEBUG $(OPT_FLAGS)
 # Compiling library source
 SYSCONF_CXXFLAGS_LIB	= -fPIC
 SYSCONF_CFLAGS_LIB	= -fPIC
@@ -165,7 +165,6 @@
 8051.o:  8051.cc 8051.h
 
 linetext.o: linetext.cc linetext.h
-	g++ -c -o linetext.o linetext.cc 
 
 debugwidget.o: debugwidget.cc \
 		debugwidget.h
